main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -1347,12 +1352,26 @@ protected void onActivityResult(final int requestCode, final int resultCode, fin
             currentFilter = FilterActivity.getFilterFromPosition(filterIndex[0], filterIndex[1]);
             setFilter();
         } else if (requestCode == GeocacheFilterActivity.REQUEST_SELECT_FILTER && resultCode == Activity.RESULT_OK) {
-            currentCacheFilter = data.getParcelableExtra(GeocacheFilterActivity.EXTRA_FILTER_CONTEXT);
-            setFilter();
-            refreshFilterForOnlineSearch();
+            setAndRefreshFilterForOnlineSearch(data.getParcelableExtra(GeocacheFilterActivity.EXTRA_FILTER_CONTEXT));
         }
     }
 
+    @Override
+    public void refreshWithFilter(final GeocacheFilter filter) {
+        currentCacheFilter.set(filter);
+        setFilter();
+        refreshFilterForOnlineSearch();
+
+        refreshCurrentList(AfterLoadAction.CHECK_IF_EMPTY);
```

Comment:
without the following call of the methods, it did not work before, have to check with the new FilterContext...